### PR TITLE
Update Lexilla and Scintilla

### PR DESCRIPTION
**Pull request by data-man**
* Lexilla to version 5.2.0
  * Scintilla to version 5.3.1
----
Notes:
* ~sources taken from repositories with important (in my opinion) post-release fixes~ 
* new Scintilla's features are not used

### CODE REVIEW

**Comment by elextr** [comment]
From a quick look there is a new lexical element SCE_P_ATTRIBUTE which needs to be mapped [here](https://github.com/geany/geany/blob/563d94d856ff5ee799f8c8505681ba4544514c33/src/highlightingmappings.h#L1352).

**Comment by kugel-** [comment]
> sources taken from repositories with important (in my opinion) post-release fixes

I would rather not add patches not found in the target scintilla release, unless there's a very good reason. If the post-release fixes are important, then lets just wait for 5.3.1

File: src/highlightingmappings.h [hunk 1]
```diff
@@ -1330,25 +1330,26 @@ static const HLKeyword highlighting_keywords_POWERSHELL[] =
...
-	{ SCE_P_WORD2,			"word2",			FALSE },
-	{ SCE_P_FSTRING,		"fstring",			FALSE },
+	{ SCE_P_WORD2,			"word2",		FALSE },
+	{ SCE_P_FSTRING,		"fstring",		FALSE },
 	{ SCE_P_FCHARACTER,		"fcharacter",		FALSE },
-	{ SCE_P_FTRIPLE,		"ftriple",			FALSE },
-	{ SCE_P_FTRIPLEDOUBLE,	"ftripledouble",	FALSE },
+	{ SCE_P_FTRIPLE,		"ftriple",		FALSE },
+	{ SCE_P_FTRIPLEDOUBLE,		"ftripledouble",	FALSE },
+	{ SCE_P_ATTRIBUTE,		"fattribute",		FALSE },
```
Comment (kugel-) [inline src/highlightingmappings.h:1352]: fattribute -> attribute, or is this intention?

File: src/highlighting.c [hunk 2]
```diff
@@ -1436,7 +1436,8 @@ gboolean highlighting_is_string_style(gint lexer, gint style)
 				style == SCE_P_FCHARACTER ||
 				style == SCE_P_FTRIPLE ||
 				style == SCE_P_FTRIPLEDOUBLE ||
-				style == SCE_P_STRINGEOL);
+				style == SCE_P_STRINGEOL ||
+				style == SCE_P_ATTRIBUTE);
```
Comment (kugel-) [inline src/highlighting.c:1440]: If I understand the usage in LexPython.cxx correctly then this new style is totally not a string, It seems to refer to Python class attributes, i.e. this is more of an identifier like class members.

**Review by kugel- — CHANGES_REQUESTED** [review]

**Comment by data-man** [comment]
Update Lexilla to 5.2.0 and Scintilla to 5.3.1 versions.

Scintilla 5.3.1, released 12 October 2022.
Lexilla 5.2.0, released 12 October 2022.

**Comment by data-man** [comment]
Closed as unnecessary after two months of ignoring it.

**Comment by elextr** [comment]
Geany is a volunteer project, nobody is working on it full time.  So just because it hasn't been actioned in some arbitrary time does not mean its unnecessary, just that nobody has had time to review, test and merge yet.

**Pull request closed without merging**